addons/account/models/account_move_line.py

```diff
@@ -669,14 +669,24 @@ def _compute_amount_currency(self):
 
     @api.depends('full_reconcile_id.name', 'matched_debit_ids', 'matched_credit_ids')
     def _compute_matching_number(self):
+        large_partials = False
         for record in self:
             if record.full_reconcile_id:
                 record.matching_number = record.full_reconcile_id.name
             elif record.matched_debit_ids or record.matched_credit_ids:
-                record.matching_number = 'P'
+                partials = record.matched_debit_ids or record.matched_credit_ids
+                if not large_partials and len(partials) > 1:
+                    large_partials = True
+                if len(partials) == 1:
+                    record.matching_number = 'P' + str(partials.id)
             else:
                 record.matching_number = None
 
+        if large_partials:
+            for record in self:
+                if record.matched_debit_ids or record.matched_credit_ids:
+                    record.matching_number = 'P*'
+
```

Comment:
I know it looks a bit dirty right now, but consider this scenario:

- there is 3 record in self
- when going through the 1st record, we see that len(partials) == 1, we assign P+ID to that
- when going through the 2nd record, we see that len(partials) > 1, we assign P*
- same for 3rd record (P*) because large_partials is already true

but at the end of this loop, the 1st record is still P+ID, we still need to update that to P*. That's why I added the 2nd loop.


Though, with my current approach we still have some problem too:
- From the bank reconciliation widget, we reconcile 2 line (say line A and B), assigning P+ID to both of them
- After that, create another reconciliation between a new line and 1 of the line from previous reconciliation (say line B and C)
- This compute method will run for that new reconciliation, but it will not include the line from first reconciliation that is not included in the new reconciliation (line A)
- Now the problem is line A still have matching_number of P+ID, and line B and line C have matching_number of P*
- What should happen is line A should also have P*

cc: @chklop @h4818

Reply:
Indeed, at this point we can consider a limitation.